### DISCUSSION TITLE
Tutorial

### DISCUSSION THREAD

**Discussion by dizzyjaguar** [wakujs/waku#620] [Ideas]
Currently in progress of starting to write out a tutorial for users to have a fun, practical, and minimalist guide to using **Waku**.

**Comment by dizzyjaguar** [comment]
@dai-shi 
My current questions for opinion are:

1.  I'm leaning towards the tutorial building a light ecommerce store. (Not sure what product yet, I'd like it to be of theme with Waku)
2. Should I include the usage of **Jotai** in the tutorial since its recommended in the docs?
3. Do you think its important to include a clone-able starter repo for the tutorial, or is that better to add on later to a v2 of the tutorial? 

I could make something relatively simplistic for now without a starter and just go through the core concepts, but make it easy to add in more steps/features down the road.

**Reply by dai-shi** [reply]
1: Sounds good
2: I don't know, tbh. If it makes sense to have it, it should be included.
3: I think what's important is the readability, but I assume people would like to have a repo that they can play with. I think you will be creating it anyway, so you can simply share it for a start. (I mean, it can be under your github profile for now.)

> I could make something relatively simplistic for now without a starter and just go through the core concepts, but make it easy to add in more steps/features down the road.

I would generally prefer to start small and add more afterwards. But we will probably announce it widely after it reaches a meaningful outcome.

**Comment by dizzyjaguar** [comment]
Working on my reference repo with a **Todo List** as I go through the docs and make note of all the features. I will then use that as reference to write out the **tutorial** for a _e-commerce_ _fruit store_ (I found some free-to-use licensed images of fruits that are simple, clean, but have a nice look) . Also will have a starter repo setup with some of the mock api functions and a few components.

Heres a list of the features I was planning on showcasing in the v1 **tutorial**
## Features for v1 tutorial:

- React Server Components / Data fetching with RSC
- Client Components
- Link components/navigation
- Pages Router
    - _layout
    - index.tsx
    - [slug]
- Meta Data

@dai-shi let me know if you think the **tutorial** should go through any other features in v1, or if that looks like a good starting point 😄

**Reply by dai-shi** [reply]
I think there can be multiple tutorials and the first one (like "getting started") could be simpler (it doesn't need to cover all features), but overall it looks good.

I wonder if @sandren has some opinions.

**Reply by dizzyjaguar** [reply]
Sounds good! I'll try to get a version 1 of it finished in the next week or 2. Then we can iterate off it.

**Comment by sandren** [comment]
> I wonder if @sandren has some opinions.

Yes, I also like the idea of having multiple tutorials with increasing complexity. 🙂

- "getting started" (Waku API + RSC basics)
- standard todo app
- ecommerce website

**Reply by dizzyjaguar** [reply]
Awesome. I already started an ecommerce and todo app! 

@sandren  For the **Getting Started** tutorial do you think it should just go over the **Waku API** & **RSC basics** _in relation_ to what `npm create waku@latest` gives the user? (files/folder wise); and possibly a few code changes?

**Reply by sandren** [reply]
> @sandren For the **Getting Started** tutorial do you think it should just go over the **Waku API** & **RSC basics** _in relation_ to what `npm create waku@latest` gives the user? (files/folder wise); and possibly a few code changes?

Yes, that would be great. Thank you! 🙂

**Comment by dizzyjaguar** [comment]
@dai-shi  & @sandren  [here](https://github.com/dai-shi/waku/pull/650) is my pull request with the **first chapter** for the tutorial

I really tried to keep this first chapter super minimal without going into _too much_. In the **second chapter** (creating a **todo app** from this initial scaffolded repo) I plan to go over the same covered topics plus a few more but all with hands on experience. Then leaving a few more of the API's for **chapter 3** too. 

Let me know any thoughts/suggestions/edits.
Thanks 😄

**Comment by dizzyjaguar** [comment]
@dai-shi & @sandren 
Sorry it's taking me forever to get on this **2nd tutorial chapter**, I got swamped with work. But it's still on my **TODO's** 😀
